app/upgrades/v0.0.10.go

```diff
@@ -0,0 +1,127 @@
+			gatewayParams := keepers.GatewayKeeper.GetParams(ctx)
+			newMinStakeGW := cosmosTypes.NewCoin("upokt", math.NewInt(1000000))
+			gatewayParams.MinStake = &newMinStakeGW
+			err = keepers.GatewayKeeper.SetParams(ctx, gatewayParams)
+			if err != nil {
+				return vm, err
+			}
+
+			// Adjust proof module parameters
+			// Validate with: `poktrolld q proof params --node=https://testnet-validated-validator-rpc.poktroll.com/`
+			//
+			// !!!! THE UPGRADE CURRENTLY BREAKS HERE !!!!
+			// https://gist.github.com/okdas/535bc3c0282483fef27454c70d889aa6
+			// proofParams := keepers.ProofKeeper.GetParams(ctx)
+			// Need to construct params manually due to protobuf type changes:
+			newProofRequirementThreshold := cosmosTypes.NewCoin("upokt", math.NewInt(20000000))
+			newProofMissingPenalty := cosmosTypes.NewCoin("upokt", math.NewInt(320000000))
+			newProofSubmissionFee := cosmosTypes.NewCoin("upokt", math.NewInt(1000000))
+			proofParams := prooftypes.NewParams(
+				0.25,
+				&newProofRequirementThreshold,
+				&newProofMissingPenalty,
+				&newProofSubmissionFee,
+			)
+
+			err = keepers.ProofKeeper.SetParams(ctx, proofParams)
+			if err != nil {
+				return vm, err
+			}
+
+			// Add new shared module params
+			// Validate with: `poktrolld q shared params --node=https://testnet-validated-validator-rpc.poktroll.com/`
+			sharedParams := keepers.SharedKeeper.GetParams(ctx)
+			sharedParams.SupplierUnbondingPeriodSessions = uint64(1)
+			sharedParams.ApplicationUnbondingPeriodSessions = uint64(1)
+			sharedParams.ComputeUnitsToTokensMultiplier = uint64(42)
+			err = keepers.SharedKeeper.SetParams(ctx, sharedParams)
+			if err != nil {
+				return vm, err
+			}
+
+			//
+			// Add new authz authorizations:
+			// https://github.com/pokt-network/poktroll/compare/v0.0.9-3...ff76430
+			//
+
+			// Validate before after with:
+			// `poktrolld q authz grants-by-granter pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t --node=https://testnet-validated-validator-rpc.poktroll.com/`
+			newAuthorizations := []grantAuthorization{
+				newTestNetGrantAuthorization("/poktroll.gateway.MsgUpdateParam"),
+				newTestNetGrantAuthorization("/poktroll.application.MsgUpdateParam"),
+				newTestNetGrantAuthorization("/poktroll.supplier.MsgUpdateParam"),
+			}
+			for _, authorization := range newAuthorizations {
+				err = keepers.AuthzKeeper.SaveGrant(
+					ctx,
+					authorization.grantee,
+					authorization.granter,
+					authorization.authorization,
+					authorization.expiration,
+				)
+				if err != nil {
+					return vm, err
+				}
+			}
+
+			// Seems like RelayMiningDifficulty have been moved from `tokenomics` to `services`.
+			// In the ideal scenario, we should have migrated the data before removing query/msg types in `tokenomics`
+			// module. It would be hard to do that now. We know that new RelayMiningDifficulty will be created,
+			// we can skip this step now.
```

Comment:
```suggestion
			// RelayMiningDifficulty have been moved from `tokenomics` to `services` in this diff.
			// Ideally (in prod), we should have migrated the data before removing query/msg types in `tokenomics` module.
			// In practice (development), we don't want to spend time on it.
			// Since we know that new RelayMiningDifficulty will be re-created and real tokens are not in danger,
			// we can skip this step now.
```